pkg/agent/util/syscall/syscall_windows.go

```diff
@@ -0,0 +1,395 @@
+	LinkLocalAddressTimeout              uint32
+	ZoneIndices                          [ScopeLevelCount]uint32
+	SitePrefixLength                     uint32
+	Metric                               uint32
+	NlMtu                                uint32
+	Connected                            bool
+	SupportsWakeUpPatterns               bool
+	SupportsNeighborDiscovery            bool
+	SupportsRouterDiscovery              bool
+	ReachableTime                        uint32
+	TransmitOffload                      NlInterfaceOffloadRodFlags
+	ReceiveOffload                       NlInterfaceOffloadRodFlags
+	DisableDefaultRoutes                 bool
+}
+
+type RawSockAddrInet struct {
+	Family uint16
+	data   [26]byte
+}
+
+func (a *RawSockAddrInet) IP() net.IP {
+	if a == nil {
+		return nil
+	}
+	if a.Family == AF_INET {
+		addr := (*syscall.RawSockaddrInet4)(unsafe.Pointer(a))
+		return net.IPv4(addr.Addr[0], addr.Addr[1], addr.Addr[2], addr.Addr[3])
+	}
+	if a.Family == AF_INET6 {
+		addr := (*syscall.RawSockaddrInet6)(unsafe.Pointer(a))
+		return addr.Addr[:]
+	}
+	return net.IPv6unspecified
+}
+
+func (a *RawSockAddrInet) String() string {
+	return a.IP().String()
+}
+
+func NewRawSockAddrInetFromIP(ip net.IP) *RawSockAddrInet {
+	sockAddrInet := new(RawSockAddrInet)
+	if ip.To4() != nil {
+		addr, _ := netip.AddrFromSlice(ip.To4())
+		addr4 := (*windows.RawSockaddrInet4)(unsafe.Pointer(sockAddrInet))
+		addr4.Family = AF_INET
+		addr4.Addr = addr.As4()
+		addr4.Port = 0
+		addr4.Zero = [8]byte{}
+		return sockAddrInet
+	}
+	addr, _ := netip.AddrFromSlice(ip)
+	addr6 := (*windows.RawSockaddrInet6)(unsafe.Pointer(sockAddrInet))
+	addr6.Family = AF_INET6
+	addr6.Addr = addr.As16()
+	addr6.Port = 0
+	addr6.Flowinfo = 0
+	scopeId := uint32(0)
+	if z := addr.Zone(); z != "" {
+		if s, err := strconv.ParseUint(z, 10, 32); err == nil {
+			scopeId = uint32(s)
+		}
+	}
+	addr6.Scope_id = scopeId
+	return sockAddrInet
+}
+
+type AddressPrefix struct {
+	Prefix       RawSockAddrInet
+	prefixLength uint8
+	_            [2]byte // Add two bytes to keep alignment.
+}
+
+func (p *AddressPrefix) IPNet() *net.IPNet {
+	if p == nil {
+		return nil
+	}
+	sockAddr := p.Prefix
+	if sockAddr.Family == AF_INET {
+		return &net.IPNet{
+			IP:   (&sockAddr).IP().To4(),
+			Mask: net.CIDRMask(int(p.prefixLength), 8*net.IPv4len),
+		}
+	}
+	if p.Prefix.Family == AF_INET6 {
+		return &net.IPNet{
+			IP:   (&sockAddr).IP(),
+			Mask: net.CIDRMask(int(p.prefixLength), 8*net.IPv6len),
+		}
+	}
+	return nil
+}
+
+func (p *AddressPrefix) EqualsTo(ipNet *net.IPNet) bool {
+	if ipNet == nil && p == nil {
+		return true
+	} else if ipNet == nil || p == nil {
+		return false
+	}
+	dstOnes, dstBits := ipNet.Mask.Size()
+	if dstOnes != int(p.prefixLength) {
+		return false
+	}
+	if p.prefixLength == 0 {
+		return ipNet.IP.Equal(net.IPv4zero) || ipNet.IP.Equal(net.IPv6zero)
+	}
+	srcNet := p.IPNet()
+	_, srcBits := srcNet.Mask.Size()
+	if srcBits != dstBits {
+		return false
+	}
```

Comment:
`utilip.IPNetEqual(p.IPNet(), ipNet)` has covered all these logic so all of them can be removed.

Reply:
Removed comparasion about mask bits, but the comparasion on prefixLength and "nil" are left.